scripts/mod_loader/modapi/game.lua

```diff
@@ -0,0 +1,177 @@
+
+local GameClass = {}
+
+GameClass.GetRep = function()
+	return CUtils.GetRep()
+end
+
+GameClass.GetPow = function()
+	return CUtils.GetPower()
+end
+
+GameClass.GetMaxPow = function()
+	return CUtils.GetMaxPower()
+end
+
+GameClass.GetCores = function()
+	return CUtils.GetCores()
+end
+
+GameClass.GetGridDef = function()
+	return CUtils.GetGridDef()
+end
+
+
+GameClass.AddRep = function(reputation)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "AddRep",
+		params = { reputation },
+		{ "number|int" }
+	}
+	
+	local current = CUtils.GetRep()
+	local new = math.max(0, current + reputation)
+	
+	CUtils.SetRep(new)
+end
+
+GameClass.SubRep = function(reputation)
+	GameClass.AddRep(-reputation)
+end
+
+GameClass.AddPow = function(power)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "AddPow",
+		params = { power },
+		{ "number|int" }
+	}
+	
+	local current = CUtils.GetPower()
+	local new = math.max(0, current + power)
+	
+	CUtils.SetPower(new)
+end
+
+GameClass.SubPow = function(power)
+	GameClass.AddPow(-power)
+end
+
+GameClass.AddMaxPow = function(power_max)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "AddMaxPow",
+		params = { power_max },
+		{ "number|int" }
+	}
+	
+	local current = CUtils.GetMaxPower()
+	local new = math.max(1, current + power_max)
+	
+	CUtils.SetMaxPower(new)
+end
+
+GameClass.SubMaxPow = function(power_max)
+	GameClass.AddMaxPow(-power_max)
+end
+
+GameClass.AddCores = function(cores)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "AddCores",
+		params = { cores },
+		{ "number|int" }
+	}
+	
+	local current = CUtils.GetCores()
+	local new = math.max(0, current + cores)
+	
+	CUtils.SetCores(new)
+end
+
+GameClass.SubCores = function(cores)
+	GameClass.AddCores(-cores)
+end
+
+GameClass.AddGridDef = function(grid_def)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "AddGridDef",
+		params = { grid_def },
+		{ "number|int" }
+	}
+	
+	local current = CUtils.GetGridDef()
+	local new = math.min(85, math.max(-15, current + grid_def))
+	
+	CUtils.SetGridDef(new)
+end
+
+GameClass.SubGridDef = function(grid_def)
+	GameClass.AddGridDef(-grid_def)
+end
+
+GameClass.GridBlink = function()
+	CUtils.DoGridBlink()
+end
```

Comment:
❗ Fails with error `bad argument #1 to 'DoGridBlink' (number expected, got no value)`

Reply:
Grid blink appears to be more involved than it first appeared, so I have removed this function.